From: Jan Kara <jack@suse.cz>
Subject: jbd2: Preserve kABI when adding j_abort_mutex
Patch-mainline: Never, kABI
References: bsc#1173833

Adding j_abort_mutex at the end is safe WRT kABI because journal_s is only
always allocated and initialized by the jbd2 layer (journal_init_common()).

Signed-off-by: Jan Kara <jack@suse.cz>

---
 include/linux/jbd2.h |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

--- a/include/linux/jbd2.h
+++ b/include/linux/jbd2.h
@@ -753,11 +753,6 @@ struct journal_s
 	int			j_errno;
 
 	/**
-	 * @j_abort_mutex: Lock the whole aborting procedure.
-	 */
-	struct mutex		j_abort_mutex;
-
-	/**
 	 * @j_sb_buffer: The first part of the superblock buffer.
 	 */
 	struct buffer_head	*j_sb_buffer;
@@ -1149,6 +1144,12 @@ struct journal_s
 	 */
 	struct lockdep_map	j_trans_commit_map;
 #endif
+#ifndef __GENKSYMS__
+	/**
+	 * @j_abort_mutex: Lock the whole aborting procedure.
+	 */
+	struct mutex		j_abort_mutex;
+#endif
 };
 
 #define jbd2_might_wait_for_commit(j) \
